server/src/main/java/org/opensearch/action/search/CreatePitResponse.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.action.search;
+
+import org.opensearch.action.ActionResponse;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.xcontent.StatusToXContentObject;
+import org.opensearch.common.xcontent.XContentBuilder;
+import org.opensearch.rest.RestStatus;
+import org.opensearch.rest.action.RestActions;
+
+import java.io.IOException;
+
+/**
+ * Create point in time response with point in time id and shard success / failures
+ */
+public class CreatePitResponse extends ActionResponse implements StatusToXContentObject {
+    // point in time id
+    private final String id;
+    private final int totalShards;
+    private final int successfulShards;
+    private final int failedShards;
+    private final int skippedShards;
+    private final ShardSearchFailure[] shardFailures;
+    private final long creationTime;
+
+    public CreatePitResponse(SearchResponse searchResponse, long creationTime) {
+        if (searchResponse.pointInTimeId() == null || searchResponse.pointInTimeId().isEmpty()) {
+            throw new IllegalArgumentException("Point in time ID is empty");
+        }
+        this.id = searchResponse.pointInTimeId();
+        this.totalShards = searchResponse.getTotalShards();
+        this.successfulShards = searchResponse.getSuccessfulShards();
+        this.failedShards = searchResponse.getFailedShards();
+        this.skippedShards = searchResponse.getSkippedShards();
+        this.shardFailures = searchResponse.getShardFailures();
```

Comment:
null might result in unintended npe , we've initialized this to empty array in case if there are no failures. thoughts?

Reply:
I am asking if this could be intialized as null by default

Reply:
I did try this change to initialize to null by default when there are failures but integ tests failed because of NPE, we will need an empty array as later this is used where length is checked.